lms/djangoapps/oauth_dispatch/dot_overrides.py

```diff
@@ -0,0 +1,41 @@
+"""
+Classes that override default django-oauth-toolkit behavior
+"""
+
+from django.contrib.auth import authenticate as django_authenticate, get_user_model
+from oauth2_provider.oauth2_validators import OAuth2Validator
+
+
+def authenticate(username, password):
+    """
+    Authenticate the user, allowing the user to identify themself either by
+    username or email
+    """
+
+    if '@' in username:
```

Comment:
2 questions regarding checking for `@` character:
1. Does the data model prevent having `@` in the username field?  If not, it's possible it may actually be someone's username, rather than email address.
2. Does the data model enforce having `@` in the email field?  If not, it's possible some entries have email fields without it.

Consider changing the logic so that it first looks up by username, and if not found, looks up by email.  That's the current implementation of https://github.com/edx/edx-oauth2-provider/blob/master/edx_oauth2_provider/forms.py#L76.